logmiddleware.go

```diff
@@ -61,15 +61,47 @@ func RequestMiddlewareLogger(logger *logrus.Logger, excludedPrefix []string) mux
 			}
 
 			Get(ctx).WithFields(logrus.Fields{
-				"method":   r.Method,
-				"url":      r.URL.String(),
-				"hostname": r.URL.Hostname(),
+				"http":	map[string]interface{}{
+					"request": map[string]interface{}{
+						"method": r.Method,
+						"userAgent": map[string]interface{}{
+							"original": r.Header.Get("user-agent"),
+						},
+					},
+				},
+				"url": map[string]interface{}{
+					"path": r.URL.String(),
+				},
+				"host": map[string]interface{}{
+					"hostname": r.URL.Hostname(),
+					"ip":  r.Header.Get("x-forwaded-for"),
+				},
 			}).Info("incoming request")
 
 			next.ServeHTTP(&myw, r.WithContext(ctx))
 
 			Get(ctx).WithFields(logrus.Fields{
-				"statusCode":   myw.statusCode,
+				"http":	map[string]interface{}{
+					"request": map[string]interface{}{
+						"method": r.Method,
+						"userAgent": map[string]interface{}{
+							"original": r.Header.Get("user-agent"),
+						},
+					},
+					"response": map[string]interface{}{
+						"statusCode": myw.statusCode,
+						"body": map[string]interface{}{
+							"bytes": myw.Length(),
```

Comment:
We should rely on `Content-Length` header, I like the approach of counting actually written bytes let's keep it as a fallback if the developer does not set `Content-Length`. 

Since as we read in Write docs the header should be added automatically: https://golang.org/src/net/http/server.go?#L124

Reply:
Writing using myw.write([]byte("prova\n")) to the response,  the "content-length" header is still not set.